validator/src/validator.rs

```diff
@@ -355,6 +385,17 @@ where
                     .get_equivocation_proofs_for_block(Self::EQUIVOCATION_PROOFS_MAX_SIZE);
                 let prev_seed = head.seed().clone();
 
+                // Get the lower timestamp necessary to compute the producer time-out.
+                let timestamp = blockchain
+                    .get_block_at(
+                        head.block_number()
+                            .saturating_sub(Self::BLOCK_DISTANCE_FROM_HEAD - 1),
+                        false,
+                        None,
+                    )
+                    .map(|block| block.timestamp())
+                    .unwrap_or_default();
```

Comment:
A timestamp of `0` looks wrong here. Perhaps get the earliest block that's later than the PoS genesis block?

Reply:
Fixed